src/utils/Graphing.ts

```diff
@@ -1,39 +1,103 @@
 import { TimeSeries, Histogram } from '../types/Metrics';
 
+type Series = {
+  name: string;
+  values: number[];
+};
+
+type Normalized = {
+  x: number[];
+  ys: Series[];
+};
+
+const compare = (a: number, b: number) => {
+  if (Math.abs(a - b) < 1000) {
```

Comment:
One second tolerance, right?

Reply:
Yes. Actually, all timestamps returned by prometheus, when several timeseries are returned, should in theory be aligned. But I prefer to have this security tolerance and not rely on this assumption too much.